src/main/api/opds.ts

```diff
@@ -66,31 +66,27 @@ export class OpdsApi {
 
     public async browse(data: any): Promise<OpdsResultView> {
         const { url } = data;
-        const opdsFeedData = await httpGet(url);
-        // let opds2Publication: OPDSPublication = null;
+        const opdsFeedData = await httpGet(url, {
+            timeout: 5000,
+        });
         let opds2Feed: OPDSFeed = null;
 
-        // This is an opds feed in version 1
-        // Convert to opds version 2
         const xmlDom = new xmldom.DOMParser().parseFromString(opdsFeedData);
-        if (!xmlDom || !xmlDom.documentElement) {
+        if (xmlDom && xmlDom.documentElement) {
+            // This is an opds feed in version 1
+            // Convert to opds version 2
+            const opds1Feed = XML.deserialize<OPDS>(xmlDom, OPDS);
+            opds2Feed = convertOpds1ToOpds2(opds1Feed);
+        } else {
+            const isEntry = xmlDom.documentElement.localName === "entry";
```

Comment:
This `else` branch is `!xmlDom || !documentElement`, which means this line of code is guaranteed to crash. Typo during code copy/paste?

Reply:
Fixed in https://github.com/readium/readium-desktop/pull/542/files/f57697dda2185e6f99e6e1f71e5badd8649ed2f6..c96d9649800af35454daf776cb15213f6182fc27